hyperspy/_signals/eds_tem.py

```diff
@@ -325,12 +330,35 @@ def quantification(self,
             navigation_mask = navigation_mask.data
         xray_lines = self.metadata.Sample.xray_lines
         composition = utils.stack(intensities)
-        composition.data = utils_eds.quantification_cliff_lorimer(
-            composition.data, kfactors=kfactors,
-            mask=navigation_mask) * 100.
+        if method == 'CL':
+            composition.data = utils_eds.quantification_cliff_lorimer(
+                composition.data, kfactors=factors,
+                mask=navigation_mask) * 100.
+        elif method == 'zeta':
+            results = utils_eds.quantification_zeta_factor(
+                composition.data, zfactors=factors,
+                dose=self._get_dose(method))
+            composition.data = results[0] * 100.
+            mass_thickness = intensities[0].deepcopy()
+            mass_thickness.data = results[1]
+            mass_thickness.metadata.General.title = 'Mass thickness'
+        elif method == 'cross_section':
+            results = utils_eds.quantification_cross_section(composition.data,
+                    cross_sections=factors,
+                    dose=self._get_dose(method))
+            composition.data = results[0] * 100
+            number_of_atoms = utils.stack(intensities)
+            number_of_atoms.data = results[1]
+            number_of_atoms = number_of_atoms.split()
+        else:
+            raise Exception ('Please specify method for quantification, as CL, zeta or cross_section')
```

Comment:
assuming the change to force specification of method has worked this exception can be removed